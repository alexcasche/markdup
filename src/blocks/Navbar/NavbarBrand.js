import { RouterLink } from 'elements'
import { shade, tint } from 'polished'

const NavbarBrand = RouterLink.extend`
  position: relative;
  bottom: 1px;
  color: ${props => props.theme.colors.grey};
  font-family: ${props => props.theme.text.fontSans};
  font-size: 23px;
  letter-spacing: 4px;
  text-transform: uppercase;
  text-decoration: none;
  transition: color 0.15s ease-in-out;
  span {
    color: ${props => shade(0.8, props.theme.colors.grey)};
    transition: inherit;
  }
  &:hover {
    color: ${props => tint(0.25, props.theme.colors.grey)};
    span {
      color: ${props => tint(0.65, props.theme.colors.grey)};
    }
  }
`

export default NavbarBrand
