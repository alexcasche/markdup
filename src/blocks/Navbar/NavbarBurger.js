import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'

const MODIFIERS = {
  open: ({ theme }) => `
    background-color: transparent;
    &:before {
      transform: rotate(-45deg) translate(-4px, 4px);
    }
    &:after {
      transform: rotate(45deg) translate(-4px, -4px);
    }
  `,
  hover: ({ theme }) => `
    background-color: ${tint(0.25, theme.colors.grey)};
    &:before, &:after {
      background-color: ${tint(0.25, theme.colors.grey)};
    }
  `,
}

const NavbarBurger = styled.span`
  position: relative;
  background-color: ${props => props.theme.colors.grey};
  width: 20px;
  height: 2px;
  transition: inherit;
  &:before,
  &:after {
    position: absolute;
    left: 0;
    top: -6px;
    content: '';
    width: inherit;
    height: inherit;
    background-color: ${props => props.theme.colors.grey};
    width: 20px;
    height: 2px;
    transition: inherit;
  }
  &:after {
    top: 6px;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default NavbarBurger
