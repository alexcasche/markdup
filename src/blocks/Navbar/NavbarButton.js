import { Button } from 'elements'

const NavbarButton = Button.extend`
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 20px;
  height: 40px;
  padding: 0;
  background-color: transparent;
`

export default NavbarButton
