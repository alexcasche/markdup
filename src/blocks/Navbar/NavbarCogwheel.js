import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styled from 'styled-components'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { tint } from 'polished'

const MODIFIERS = {
  open: ({ theme }) => `
    transform: rotate(45deg);
    color: ${tint(0.25, theme.colors.grey)};
  `,
  hover: ({ theme }) => `
    color: ${tint(0.25, theme.colors.grey)};
  `,
}

const NavbarCogwheel = styled(FontAwesomeIcon)`
  color: ${props => props.theme.colors.grey};
  font-size: 20px;
  transition: inherit;
  transform-origin: 50%;
  ${applyStyleModifiers(MODIFIERS)};
`

export default NavbarCogwheel
