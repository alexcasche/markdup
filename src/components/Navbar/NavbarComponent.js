import React, { Component } from 'react'
import { PropTypes as t } from 'prop-types'
import { BrowserRouter } from 'react-router-dom'
import { Navbar } from 'blocks'
import { faCog } from '@fortawesome/free-solid-svg-icons'

export default class NavbarComponent extends Component {
  constructor() {
    super()
    this.state = {
      burgerHover: false,
      cogwheelHover: false,
    }
  }
  buttonHover = button => {
    const stateKey = `${button}Hover`
    let newState = {}
    newState[stateKey] = !this.state[stateKey]
    this.setState(prevState => ({
      ...prevState,
      ...newState,
    }))
  }
  buttonModifiers = button => {
    const stateKey = `${button}Hover`
    const propKey = button === 'burger' ? 'showSidebar' : 'showSettings'
    return [
      ...(this.state[stateKey] ? ['hover'] : []),
      ...(this.props[propKey] ? ['open'] : []),
    ]
  }
  render() {
    const { actions } = this.props
    return (
      <Navbar className="Navbar">
        <Navbar.Button
          onClick={() => actions.toggleLayout('showSidebar')}
          onMouseEnter={() => this.buttonHover('burger')}
          onMouseLeave={() => this.buttonHover('burger')}
        >
          <Navbar.Burger modifiers={this.buttonModifiers('burger')} />
        </Navbar.Button>
        <BrowserRouter>
          <Navbar.Brand to="/">
            <span>markd</span>up
          </Navbar.Brand>
        </BrowserRouter>
        <Navbar.Button
          onClick={() => actions.toggleLayout('showSettings')}
          onMouseEnter={() => this.buttonHover('cogwheel')}
          onMouseLeave={() => this.buttonHover('cogwheel')}
        >
          <Navbar.Cogwheel
            icon={faCog}
            modifiers={this.buttonModifiers('cogwheel')}
          />
        </Navbar.Button>
      </Navbar>
    )
  }
}

NavbarComponent.propTypes = {
  showSettings: t.bool.isRequired,
  showSidebar: t.bool.isRequired,
  actions: t.shape({
    toggleLayout: t.func.isRequired,
  }).isRequired,
}
