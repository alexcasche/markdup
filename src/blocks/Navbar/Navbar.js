import styled from 'styled-components'

import NavbarBrand from './NavbarBrand'
import NavbarButton from './NavbarButton'
import NavbarBurger from './NavbarBurger'
import NavbarCogwheel from './NavbarCogwheel'

const Navbar = styled.div`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 50px;
  min-height: 50px;
  padding: 0 20px;
  background-color: ${props => props.theme.colors.black};
`

Navbar.Brand = NavbarBrand
Navbar.Burger = NavbarBurger
Navbar.Button = NavbarButton
Navbar.Cogwheel = NavbarCogwheel

export default Navbar
