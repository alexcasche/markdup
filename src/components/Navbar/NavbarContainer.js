import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getLayoutVal } from 'selectors'
import { toggleLayout } from 'actions'
import NavbarComponent from './NavbarComponent'

export const NavbarContainer = props => {
  return <NavbarComponent {...props} />
}

function mapStateToProps(state) {
  return {
    showSettings: getLayoutVal(state, 'showSettings'),
    showSidebar: getLayoutVal(state, 'showSidebar'),
  }
}
function mapDispatchToProps(dispatch) {
  return {
    actions: {
      toggleLayout: bindActionCreators(toggleLayout, dispatch),
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavbarContainer)
