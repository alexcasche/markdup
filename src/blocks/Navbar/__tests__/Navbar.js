import Navbar from '../Navbar'

describe('Navbar Block', () => {
  it('renders correctly', () => {
    const tree = render(<Navbar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
