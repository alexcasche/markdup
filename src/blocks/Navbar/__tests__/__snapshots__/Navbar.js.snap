// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Block renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
  min-height: 50px;
  padding: 0 20px;
  background-color: #1D1F21;
}

<div
  className="c0"
/>
`;
